src/test/java/org/apache/sysds/test/functions/io/libsvm/WriteLIBSVMTest.java

```diff
@@ -27,70 +27,70 @@
-  }
+	@Test public void testlibsvm2_Pllel_CP() {
+		runWriteLIBSVMTest(getId(), ExecMode.SINGLE_NODE, true, " ", ":", false);
+	}
 
-  @Test public void testlibsvm1_SP() {
-    runWriteLIBSVMTest(getId(), ExecMode.SPARK, false, " ", ":", true);
-  }
+	@Test public void testlibsvm1_SP() {
+		runWriteLIBSVMTest(getId(), ExecMode.SPARK, false, " ", ":", true);
+	}
 
-  @Test public void testlibsvm2_SP() {
-    runWriteLIBSVMTest(getId(), ExecMode.SPARK, false, " ", ":", false);
-  }
+	@Test public void testlibsvm2_SP() {
+		runWriteLIBSVMTest(getId(), ExecMode.SPARK, false, " ", ":", false);
+	}
 
-  protected void runWriteLIBSVMTest(int testNumber, ExecMode platform, boolean parallel, String sep, String indSep,
-    boolean sparse) {
+	protected void runWriteLIBSVMTest(int testNumber, ExecMode platform, boolean parallel, String sep, String indSep,
+		boolean sparse) {
 
-    ExecMode oldPlatform = rtplatform;
-    rtplatform = platform;
+		ExecMode oldPlatform = rtplatform;
+		rtplatform = platform;
 
-    boolean sparkConfigOld = DMLScript.USE_LOCAL_SPARK_CONFIG;
-    if(rtplatform == ExecMode.SPARK)
-      DMLScript.USE_LOCAL_SPARK_CONFIG = true;
+		boolean sparkConfigOld = DMLScript.USE_LOCAL_SPARK_CONFIG;
+		if(rtplatform == ExecMode.SPARK)
+			DMLScript.USE_LOCAL_SPARK_CONFIG = true;
 
-    boolean oldpar = CompilerConfig.FLAG_PARREADWRITE_TEXT;
+		boolean oldpar = CompilerConfig.FLAG_PARREADWRITE_TEXT;
 
-    try {
+		try {
 
-      CompilerConfig.FLAG_PARREADWRITE_TEXT = parallel;
+			CompilerConfig.FLAG_PARREADWRITE_TEXT = parallel;
 
-      TestConfiguration config = getTestConfiguration(getTestName());
-      loadTestConfiguration(config);
+			TestConfiguration config = getTestConfiguration(getTestName());
+			loadTestConfiguration(config);
 
-      String HOME = SCRIPT_DIR + TEST_DIR;
-      String inputMatrixName = HOME + INPUT_DIR + getInputLIBSVMFileName();
-      String dmlOutput = output("dml.scalar");
-      String libsvmOutputName = output("libsvm_write" + testNumber + ".data");
+			String HOME = SCRIPT_DIR + TEST_DIR;
+			String inputMatrixName = HOME + INPUT_DIR + getInputLIBSVMFileName();
+			String dmlOutput = output("dml.scalar");
+			String libsvmOutputName = output("libsvm_write" + testNumber + ".data");
 
-      fullDMLScriptName = HOME + getTestName() + "_" + testNumber + ".dml";
-      programArgs = new String[] {"-args", inputMatrixName, dmlOutput, libsvmOutputName, sep, indSep,
-        Boolean.toString(sparse)};
+			fullDMLScriptName = HOME + getTestName() + "_" + testNumber + ".dml";
+			programArgs = new String[] {"-args", inputMatrixName, dmlOutput, libsvmOutputName, sep, indSep,
+				Boolean.toString(sparse)};
 
-      runTest(true, false, null, -1);
-    }
-    finally {
-      rtplatform = oldPlatform;
-      CompilerConfig.FLAG_PARREADWRITE_TEXT = oldpar;
-      DMLScript.USE_LOCAL_SPARK_CONFIG = sparkConfigOld;
-    }
-  }
+			runTest(true, false, null, -1);
+		}
```

Comment:
Can you please add a way to verify if the written data is correct? You can read the output file in a R script and compare the sums. You can follow the example of WriteCSVTest.

Reply:
Tests added to verify read/write. Thank!.